atlasdb-config/src/test/java/com/palantir/atlasdb/http/ExceptionRetryBehaviourTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.http;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Instant;
+import java.util.Date;
+
+import org.junit.Test;
+
+import com.palantir.common.remoting.ServiceNotAvailableException;
+import com.palantir.lock.remoting.BlockingTimeoutException;
+
+import feign.RetryableException;
+
+@SuppressWarnings("ThrowableResultOfMethodCallIgnored") // We need to create exceptions for testing.
+public class ExceptionRetryBehaviourTest {
+    private static final ServiceNotAvailableException SERVICE_NOT_AVAILABLE_EXCEPTION
+            = new ServiceNotAvailableException("foo");
+    private static final BlockingTimeoutException BLOCKING_TIMEOUT_EXCEPTION = new BlockingTimeoutException("foo");
+    private static final Date DATE = Date.from(Instant.EPOCH);
+
+    @Test
+    public void retryOnOtherNodesShouldRetryInfinitelyManyTimes() {
```

Comment:
Actually, if I'm not wrong we would give up after fifteen attempts if it was false. (Maybe this is covered by the comment on documenting what that means earlier?)